src/test/java/com/alibaba/druid/util/StringUtilsTest.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright 1999-2019 Alibaba Group Holding Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.alibaba.druid.util;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+public class StringUtilsTest {
+
+    @Test
+    public void testSubStringToInteger() {
+        Assert.assertNull(StringUtils.subStringToInteger("foobar", "1", "3"));
+
+        Assert.assertEquals(new Integer(2),
+                StringUtils.subStringToInteger("1234", "1", "3"));
+    }
+
+    @Test
+    public void testSubString() {
+        Assert.assertNull(StringUtils.subString(",", "foo", ","));
+        Assert.assertNull(
+                StringUtils.subString("foo", "foo", "a\'b\'c", false));
+        Assert.assertNull(
+                StringUtils.subString("foo", "foo", "a\'b\'c", true));
+
+        Assert.assertEquals("a 2b ",
+                StringUtils.subString("1a 2b 3c", "1", "3"));
+        Assert.assertEquals("c",
+                StringUtils.subString("abcdef", "b", "d", true));
+    }
+
+    @Test
+    public void testStringToInteger() {
+        Assert.assertNull(StringUtils.stringToInteger(""));
+        Assert.assertNull(StringUtils.stringToInteger(null));
+        Assert.assertNull(StringUtils.stringToInteger("a"));
+
+        Assert.assertEquals(new Integer(3), StringUtils.stringToInteger("3"));
+    }
+
+    @Test
+    public void testEquals() {
+        Assert.assertFalse(StringUtils.equals(null, ""));
+        Assert.assertFalse(StringUtils.equals("foo", "bar"));
+
+        Assert.assertTrue(StringUtils.equals(null, null));
+        Assert.assertTrue(StringUtils.equals("foo", "foo"));
+    }
+
+    @Test
+    public void testEqualsIgnoreCase() {
+        Assert.assertFalse(StringUtils.equalsIgnoreCase(null, ""));
+        Assert.assertFalse(StringUtils.equalsIgnoreCase("foo", "bar"));
+
+        Assert.assertTrue(StringUtils.equalsIgnoreCase(null, null));
+        Assert.assertTrue(StringUtils.equalsIgnoreCase("foo", "foo"));
```

Comment:
Add `assertTrue` test with two strings that are the same letters but different case

Reply:
Done